Bug fixes on Subscription termination date format [Fixes Issues #9186]

tim1234ltp opened the pull request:
Fixes https://github.com/HabitRPG/habitica/issues/9186

### Changes
Created a computed value of Subscription termination date which is formatted according to user preference.
The client will show this computed value instead of using a filter on raw value.

### User ID:a547cd88-24e5-46d0-bd1b-eb453f653ffe

### Result (sorry for the non-english strings)
![default](https://user-images.githubusercontent.com/26185175/32132616-5b7ee01a-bbf9-11e7-9a37-99dc79cf9fbf.jpg)

tim1234ltp committed:
Bug fixes on Subscription termination date format.

tim1234ltp committed:
Bug fixes on Subscription termination date format.

tim1234ltp committed:
Solved Indent Issues.

tim1234ltp committed:
Solved Indent Issues.

tim1234ltp committed:
Bug fixes on Subscription termination date format. (sync with the upstream)

tim1234ltp committed:
Merge branch 'develop' of https://github.com/tim1234ltp/habitica into develop

Rebasing.

Alys commented:
@tim1234ltp Regarding the test failure on this PR (from your comment at https://github.com/HabitRPG/habitica/issues/9186#issuecomment-340328253), the same failure occurs when I run the test locally on a copy of `develop`, so this failure isn't coming from your code. I'm commenting here rather than in the issue so that we see this when we're reviewing your PR.

$ mocha test/api/v3/unit/libs/payments/group-plans/group-payments-create.test.js --require test/helpers/start-server.js 
```
....
  27 passing (9s)
  2 pending
  1 failing

  1) Purchasing a group plan for group adds months to members with existing recurring subscription (Amazon):
     AssertionError: expected 4.037242907706093 to be within 3..4
      at _callee16$ (test/api/v3/unit/libs/payments/group-plans/group-payments-create.test.js:478:55)
      at tryCatch (node_modules/regenerator-runtime/runtime.js:65:40)
      at GeneratorFunctionPrototype.invoke [as _invoke] (node_modules/regenerator-runtime/runtime.js:303:22)
      at GeneratorFunctionPrototype.prototype.(anonymous function) [as next] (node_modules/regenerator-runtime/runtime.js:117:21)
      at GeneratorFunctionPrototype.tryCatcher (node_modules/bluebird/js/release/util.js:16:23)
      at PromiseSpawn._promiseFulfilled (node_modules/bluebird/js/release/generators.js:97:49)
      at Promise._settlePromise (node_modules/bluebird/js/release/promise.js:574:26)
      at Promise._settlePromise0 (node_modules/bluebird/js/release/promise.js:614:10)
      at Promise._settlePromises (node_modules/bluebird/js/release/promise.js:693:18)
      at Async._drainQueue (node_modules/bluebird/js/release/async.js:133:16)
      at Async._drainQueues (node_modules/bluebird/js/release/async.js:143:10)
      at Immediate.Async.drainQueues (node_modules/bluebird/js/release/async.js:17:14)
```

SabreCat committed:
feat(event): Habitoween

SabreCat committed:
chore(sprites): compile

SabreCat committed:
chore(i18n): update locales

SabreCat committed:
4.7.0

SabreCat committed:
Merge branch 'release' into develop

SabreCat committed:
fix(sprites): bad shirt centering

SabreCat committed:
4.7.1

SabreCat committed:
Merge branch 'release' into develop

Alys committed:
fix an inaccurate comment about guilds needing summaries

Alys committed:
update github templates for new website and wiki page locations

TheHollidayInn committed:
Bulk stats (#9260)

* Reorganized stats

* Organized allocation common code

* Added bulk allocate to common

* Added allocate bulk route

* Fixed structure and lint

* Fixed import and apidoc

SabreCat committed:
4.8.0

SabreCat committed:
Merge branch 'release' into develop

TheHollidayInn committed:
Added fix for multiple level up (#9330)

TheHollidayInn committed:
Added reward text (#9332)

TheHollidayInn committed:
Prevented challenge prize edit. Fixed edit to create modal (#9374)

TheHollidayInn committed:
Fixed member modal search (#9375)

TheHollidayInn committed:
Prevented challenge edit during RYA (#9373)

paglias committed:
Tasks v2 Part 2 (#9236)

* start updating colors for tasks controls

* finish updating controls colors

* change hoevr behavior

* change transition duration

* update color with transition

* refactor menu wip

* wip

* upgrade vue deps

* fix warnings

* fix menu

* misc fixes

* more fixes

* fix badge

* fix margins in menu

* wip tooltips

* tooltips

* fix checklist colors

* add badges

* fix quick add input

* add transition to task controls too

* add batch add

* fix task filtering

* finish tasks badges

* fix menu

* upgrade deps

* fix shop items using all the same image

* fix animation

* disable client tests until we remove phantomjs

* revert changes to tasks colors

* fix opacity in task modal inputs

* remove client unit tests from travis

* wip task dropdown

* fix z-index for task footer/header

* fixes and add files

* fixes

* bigger clickable area

* more space to open task dropdown

* droddown position

* fix menu position

* make sure other dropdowns get closed correctly

* fixes

* start to fix z-index

* draggable = false for task dropdown

* fix for dropdown position

* implement move to top / bottom

* fix push to bottom

* typo

* fix drag and drop

* use standard code

* wider click area for dropdown

* unify badge look

* fix padding

* misc fixes

* more fixes

* make dropdown scrollable

* misc fixes

* fix padding for notes

* use existing code instead of new props

TheHollidayInn committed:
Add close to tags popup (#9376)

TheHollidayInn committed:
Ensured user selects plan (#9378)

phillipthelen committed:
Add "stats" as a tutorial step (#9377)

This is used mostly for the mobile apps to identify if the tutorial step about stats has be shown yet or not.

TheHollidayInn committed:
Added extra months to account for months with larger amount of days (#9379)

SabreCat committed:
feat(content): Backgrounds and Armoire 2017-11
End Habitoween and Fall Festival

tim1234ltp commented:
@Alys Thanks for the update. I would leave this PR open and wait for the review.

SabreCat committed:
fix(news): proper Take This announcement

SabreCat committed:
chore(sprites): compile

SabreCat committed:
chore(i18n): update locales

SabreCat committed:
4.9.0

TheHollidayInn committed:
Added broken task event

TheHollidayInn committed:
Removed old state item

TheHollidayInn committed:
Reloaded completed todos if we are unlinking a todo

paglias committed:
fix customize-options sprites

SabreCat committed:
Merge branch 'paglias/fix-sprites' into release

TheHollidayInn committed:
Added exitence checks (#9383)

TheHollidayInn committed:
Ensured sort value is true for checklists (#9386)

SabreCat committed:
4.9.1

paglias committed:
remove forked vue version

paglias committed:
remove forked vue version

Alys committed:
allow "visit the stable" text to be translated

Alys committed:
improve wording for Brutal Smash skill

Alys committed:
adjust login messages to indicate case-sensitivity and hint about google sign-in

paglias committed:
Fix guilds fetching (#9416)

* fix guilds fetching

* fix missing categories

paglias committed:
cache sprites and fix images caching (#9422)

asifmallik committed:
Implements repeat every X days since last completion  (Fixes #6941) (#8962)

* Implemented repeat after completion

* Added tests for repeat after completion in shouldDo.test.js

* Remove lastTicked

* Undoes removal of website/client/README.md

TheHollidayInn committed:
Updated avatar menu icons and style (#9409)

paglias committed:
fix confetti image on home page

TheHollidayInn committed:
[WIP] Add initial fixes for concurrency (#9321)

* Add initial fixes for concurrency

* Added memory edit for notifications

* Fixed tag delete

* Fixed adding and moving task order

* Updated delete task

* Fixed lint

* Fixed task adding

* Switch to mongoose push and pull

SabreCat committed:
feat(content): change to Thunderstorm Hatching Potions

SabreCat committed:
chore(i18n): update locales

SabreCat committed:
4.10.0

paglias committed:
remove unused loggin for ios purchases

negue committed:
[WIP] bootstrap-vue upgrade (#9178)

* update bootstrap-vue to 1.0.0-beta.9 - remove all individual bootstrap components and use BootstrapVue into Vue

* change modal action names from show::modal to bv::show::modal

* check if drops are undefined

* fix modal widths - sellModal now using input instead of dropbox

* upgrade to bootstrap 4.0beta

* include package-lock changes

* fix app menu dropdown position

* upgrade bootstrap to beta2 (was missing grid offset and other fixes) - refix header menu position

* fix tags popup (auto width to max not working) - fix filter panel width (adding width: 100% works until max-width)

* show hide logo on different screensize (new css breakpoints - http://getbootstrap.com/docs/4.0/utilities/display/ )

* fix package-lock?

* fix active button style / app header toggle button

* fix package-lock !

* update package lock after merge - new mixin "openedItemRows" to save the "show more/show less" in stable

* mixin naming style

* fix buyQuestModal marginTop

* fix customMenuDropdown position

* fix userDropdown items

paglias committed:
update package-lock.json

paglias committed:
chore(npm): upgrade bootstrap-vue

paglias committed:
intro tour: fix justin position

Alys committed:
add missing removeInvite string

Alys committed:
add missing removeInvite string

Alys committed:
add missing string for dateEndNovember

Alys committed:
add missing string for dateEndNovember

SabreCat committed:
Merge branch 'release' into develop

SabreCat committed:
fix(sprites): missing Guild image

SabreCat committed:
chore(i18n): update locales

SabreCat committed:
Merge branch 'release' into develop

paglias committed:
Fix issues with Bootstrap Vue and Boostrap upgrades (#9452)

* start to fix modals

* fixed cards paddings

* fix notifications not being marked as read

* add tests for reading a notification

* fixed indentation and added tests for reading multiple notifications

* register from home page using enter key

paglias committed:
Tasks: fixes and new edit task design (#9442)

* add missing tooltips

* makes sure due date is update correctly, fixes #9436

* do not collapse checklist when casting spells, fixes #9345

* start to fix spells drawer

* fix drawer requiring two clicks to open

Alys committed:
swap achievement badge icons for Challenges Won and Quests

SabreCat committed:
4.10.1

SabreCat committed:
Revert "Implements repeat every X days since last completion  (Fixes #6941) (#8962)"

This reverts commit 9d69d4b86322c8e52e607d9c8cea919f3100257d.

paglias committed:
remove outline from focused inputs

paglias committed:
fix modal handling

paglias committed:
fix checklist deleting

paglias committed:
fix bailey css

klezt committed:
Solved Issue #9334. Added unpin feature for rewards list in tasks.

klezt committed:
Does not reuse same css classes, imports them from pin.scss. For the togglePin function the notification.

TheHollidayInn committed:
Group plan fixes (#9437)

* Prevented title editing on personal page

* Fixed claim/unlclaim from user task page

* Removed task from local on delete

* Immediately show unassigned bar

* Add move to group tasks

* Fixed group member count increase

* Added upgrade when group plan is canceled

TheHollidayInn committed:
Added check for balance with respect to quantity (#9469)

klezt committed:
Added pin.scss to index.scss.

SabreCat committed:
feat(content): Yarn Pet Quest

SabreCat committed:
chore(sprites): compile

SabreCat committed:
chore(i18n): update locales

SabreCat committed:
4.11.0

SabreCat committed:
Merge branch 'release' into develop

klezt committed:
Deleted unnessecary import.

SabreCat committed:
chore(sprites): recompile

paglias commented on website/client/components/settings/subscription.vue:
```diff
@@ -237,6 +237,10 @@ export default {
       };
     },
+    dateTerminated () {
+      if (!this.user.preferences || !this.user.preferences.dateFormat) return moment(this.user.purchased.plan.dateTerminated);
```
this should return a string not a moment object, calling `.format()` should be enough

paglias reviewed (changes requested):
Other than the comment I left this looks good, sorry for the delay!

paglias committed:
More Tasks page fixes (#9475)

* fix datepicker not closing when clicking outside of it, fixes #9346

* fixes #9441

crookedneighbor committed:
Rename constant for restricted email domains (#9459)

zaroth committed:
added favicons to the new client (#9235)

thefridge111 committed:
Changes animal attribute that A-Z sort is applied to. (#9241)

aquibm committed:
Modify relative dueIn time on tasks to be in days (#9251)

* Modify relative dueIn time on tasks to be in days

- Normalizes the current time and task due time to the ends of their respective days.
- Returns 'today' if the dates are the same day else uses moment's humanize function to allow for weeks, months, years and so on.

* Modify task due date to appear grey when due the next day

rholinshead committed:
Choose class modal remove tooltip (#9298)

* Change chooseClass modal opt out button and tooltip
- Remove hover tooltip from opt out
- Fix centering of choose class button
- Add some margin between opt out and choose class buttons
- Make cursor on opt out text the pointer to make it obvious it's clickable

* Alphabetical order class selector

EmilyOng committed:
changed to 'createChallengeCloneTasks' (#9300)

* changed to 'createChallengeCloneTasks'

* createChallengeCloneTasks

* Added new i18n strings

dnlup committed:
Fix Incorrect Heading margins (#9311)

* fix(style): Fix margin of headers in inventory page

Set the `mb-0` class in h1 header to `mb-4`.
Set the class if `h2` header to `mb-3`.

* fix(style): Fix margin of headers in shops section

Set the `mb-0` class in h1 header to `mb-4`.
Set the class of `h2` header to `mb-3`.

* fix(style): Fix margin of headers in shops adn time traveler section

Set the `mb-0` class in h1 header to `mb-4`.
Set the class of `h2` header to `mb-3`.

Pugzy committed:
Update appFooter.vue in attempt to fix #9336 (#9348)

* Update appFooter.vue in attempt to fix #9336

Make whole social circle clickable.
Replace div with `a` tag allowing whole circle to become a link.

* Indent and reformat "Not ready yet" Instagram button.

stephqwu committed:
#9354 Indicate that Orb of Rebirth is instant (#9357)

* #9354 Indicate that Orb of Rebirth is instant

Clarified the text

* #9354 Indicate that Orb of Rebirth is instant

Reverted changes in all but locales/en

rcmbal committed:
Removed "Tag list in tasks starts collapsed" settings and strings (#9406)

Mathwhiz committed:
Fix regex for chat highlighting (#9411)

* Fix regex for chat highlighting

* Fix lint errors

AlyxMoon committed:
Add variable placeholder. (#9465)

Changes the character translation string `charactersRemaining`  in
groups.json to include a variable placeholder `<%= characters %>`

paglias committed:
update package-lock.json

paglias committed:
Merge branch 'issue9334' of https://github.com/CSE2410-TeamZero/habitica into CSE2410-TeamZero-issue9334

paglias committed:
Merge branch 'CSE2410-TeamZero-issue9334' into develop

Alys committed:
remove client-side purchase of Armoire - fixes #9432 (#9463)

Mathwhiz committed:
Add quantity information when purchasing items (#9481)

Alys committed:
remove unneeded space from in front of a full stop

SabreCat committed:
fix(drawer): remove overflow CSS

Alys committed:
allow banned word / swearword blocker to apply to most public guilds (#9253)

* allow banned word / swearword blocker to apply to all public guilds, with specified exceptions

* add another guild

* add more guilds to those that do not have the bannedWords blocker applied

* fix lint errors

Mathwhiz committed:
Fix regex for message highlighting (#9516)

SabreCat committed:
fix(notifications): extend snackbar duration

paglias committed:
fix: correctly handle bootstrap variables

paglias committed:
fix Remove Claim positioning

SabreCat committed:
fix(group-plans): more claim/unclaim alignment

SabreCat committed:
chore(i18n): update locales

SabreCat committed:
4.11.1

TheHollidayInn committed:
Hot fixes nov 15 (#9519)

* Added fortify sync. Removed multiple buy

* Added yesterdaily check

* Fixed checks for running yesterdailies

TheHollidayInn committed:
Group plan fixes (#9518)

* Fixed group plan editing

* Added translations

* Abstracted query for group or challenge tasks

Mathwhiz committed:
Disable immutable inputs when editing a challenge (#9412)

* Disable uneditable inputs when editing a challenge

* Revert prize display when creating

SabreCat committed:
fix(seasonal-shop): short circuit if categories empty

SabreCat committed:
feat(event): Turkey Day 2017
and (content) November Subscriber Items

SabreCat committed:
chore(sprites): compile

SabreCat committed:
chore(i18n): update locales

SabreCat committed:
4.12.0

SabreCat committed:
Merge branch 'release' into develop

TheHollidayInn committed:
Challenge fixes (#9528)

* Added challenge member search to progress dropdown

* Added leave challenge modal

* Allowed editing for challenge leader only

* Pevented users from editing challenge task info

* Set default progress default to daily

* Removed reward filters from user challenge progress

TheHollidayInn committed:
Added account transfer migration (#9548)

* Added account transfer migration

* Removed bad comment

TheHollidayInn committed:
Should do diff week fix (#9551)

* Fixed week difference when changing year

* Added year switchover test

* Fixed start date setting

TheHollidayInn committed:
Cancel users free group plan when they leave a group (#9543)

* Cancel users free group plan when they leave a group

* Fixed lint

TheHollidayInn committed:
Fixed display of RYA behind bailey (#9555)

SabreCat committed:
chore(i18n): update locales

SabreCat committed:
4.12.1

SabreCat committed:
Merge branch 'release' into develop

paglias committed:
fix navbar not showing up in homepage

Alys committed:
remove words related to alcohol

This is because they're causing problems in housework guilds (alcohol
is used for cleaning) and since it's coming up to Christmas a lot of
guilds are starting to have permissible conversations about drinks,
rum balls, etc.

We might put these back after Christmas but make specific exceptions
for the housework guilds.

When a more advanced word blocker is introduced, we'll be able to
ban alcohol words only in the Tavern. For irony.

tim1234ltp commented:
I will try to fix the PR this weekend.

tim1234ltp committed:
Rebase and fixed the return value

tim1234ltp committed:
Resolved merge conflict

tim1234ltp commented:
Oops when I updated my repo with the newest one, tons of commit just generated...
Please kindly ignore them.

tim1234ltp committed:
Fixed return value

Alys commented:
Hi @tim1234ltp! These things happen. :) We will need to have the extra commits removed so that we can accurately review your change, but if you're not sure how to do that, just post here and we'll give you some advice!

tim1234ltp commented:
@Alys Which commits? Those not made by me? Should I just set up a clean repo and start from that?

paglias commented:
@tim1234ltp it looks like a lot of extraneous commits got included so it's probably a good idea to start a new clean branch forked from `develop` and re-apply your changes there

tim1234ltp commented:
I will start a new PR with clean branch and close this PR.
